twisted/internet/_sslverify.py

```diff
@@ -1645,11 +1648,11 @@ def _verifyCallback(conn, cert, errno, depth, preverify_ok):
             name = "%s-%d" % (reflect.qual(self.__class__), _sessionCounter())
             sessionName = md5(networkString(name)).hexdigest()
 
-            ctx.set_session_id(sessionName)
+            ctx.set_session_id(sessionName.encode('ascii'))
```

Comment:
@acabhishek942 your change is fine.  In the hashlib library, the hexdigest() method returns a **str**:
https://github.com/python/cpython/blob/master/Modules/clinic/md5module.c.h#L41